### PR TITLE
Build recipe to clone Intel ICC runtime.

### PR DESCRIPTION
**Pull request by stuartarchibald**
This recipe clones the Intel ICC Runtime (icc_rt) to a new package
that has no "features".

### CODE REVIEW

**Comment by stuartarchibald** [comment]
Not tested anywhere but linux yet.

**Comment by seibert** [comment]
I can't get this recipe to work on Windows, for a couple reasons, some small and some large:

* The Windows build script should be called `bld.bat`, not `build.bat` (easy fix)
* The conda executable is not located in `$SYS_PREFIX/bin/conda` but rather `$SYS_PREFIX/Scripts/conda`. (easy fix)
* If you try to launch that command with subprocess directly, you get an error that it is not a Win32 application. (?)
* If you instead try to launch cmd.exe, then the build seems to hang?

I'm thinking it might be easier to do this portably if we let conda-build fetch the Intel package directly using the URL: https://anaconda.org/intel/icc_rt/2018.0.2/download/win-64/icc_rt-2018.0.2-intel_0.tar.bz2

**Comment by mbargull** [comment]
You probably just want to patch the package index entry directly to save you the trouble.
I happen to have this little script:
<details><summary><code>
patch_no_features.py
</code></summary>

```python
#!/usr/bin/env python

from argparse import ArgumentParser
from io import BytesIO
from json import load as json_load, dumps as json_dumps
from tarfile import open as tar_open
from time import time


def patch_index(src_index_file, build_string_suffix):
    index = json_load(src_index_file)
    index["build"] = "".join((index["build"], build_string_suffix))
    index.pop("features", None)
    index.pop("track_features", None)

    index_str = json_dumps(index, indent=2, sort_keys=True)
    index_bytes = index_str.encode("ascii")
    index_file = BytesIO()
    index_file.write(index_bytes)
    size = index_file.tell()
    index_file.seek(0)
    return index_file, size


def patch_package(src_path, build_string_suffix):
    ext = ".tar.bz2"
    dst_path = "".join((src_path.rstrip(ext), build_string_suffix, ext))
    with tar_open(src_path, "r") as src:
        with tar_open(dst_path, "w:bz2") as dst:
            for info in src:
                data = src.extractfile(info)
                if info.name == "info/index.json":
                    data, size = patch_index(data, build_string_suffix)
                    info.size = size
                    info.mtime = int(time())
                dst.addfile(info, data)


def get_arg_parser():
    arg_parser = ArgumentParser()
    add = arg_parser.add_argument

    add(
        "files",
        nargs="+",
    )
    add(
        "--suffix",
        default="_no_features",
    )

    return arg_parser


def main(args=None):
    arg_parser = get_arg_parser()
    args = arg_parser.parse_args()
    for package_file in args.files:
        patch_package(package_file, args.suffix)


if __name__ == "__main__":
    main()
```
</details>

**Comment by mbargull** [comment]
Instead of invoking `conda-build` you just have to use a shell/python script to download the package tarballs and run the aforementioned script on those, e.g., `patch_no_features.py /path/to/tarballs/*.tar.bz2`.

**Comment by stuartarchibald** [comment]
@mbargull thanks for the script etc. To integrate with our existing build infrastructure I've gone with conda build, this also means that what is being mutated is noted in version control and the source packages are automatically resolved and downloaded.

File: buildscripts/condarecipe_clone_icc_rt/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  name: icc_rt
+  version: {{ version }}
+
+build:
+  number: {{ build_number }}
+
+source:
+    - url: https://anaconda.org/intel/icc_rt/{{ version }}/download/win-64/icc_rt-{{version}}-intel_{{build_number}}.tar.bz2 # [win]
+    - url: https://anaconda.org/intel/icc_rt/{{ version }}/download/osx-64/icc_rt-{{version}}-intel_{{build_number}}.tar.bz2 # [osx]
+    - url: https://anaconda.org/intel/icc_rt/{{ version }}/download/linux-64/icc_rt-{{version}}-intel_{{build_number}}.tar.bz2 # [linux]
```
Comment (seibert) [inline buildscripts/condarecipe_clone_icc_rt/meta.yaml]: I know it is pedantic, but presumably we need to put a SHA256 for each of these?

Comment (stuartarchibald) [reply buildscripts/condarecipe_clone_icc_rt/meta.yaml]: I think we can, or if we don't it just downloads them fresh. Does SHA work with selectors (am going to assume so).

Comment (seibert) [reply buildscripts/condarecipe_clone_icc_rt/meta.yaml]: yes, selectors should work with any of the config lines, I think

Comment (stuartarchibald) [reply buildscripts/condarecipe_clone_icc_rt/meta.yaml]: done in https://github.com/numba/numba/pull/2881/commits/976fcd61ff191523c471f9535db05c24b882fc6a

**Comment by seibert** [comment]
ok, not going to wait for Travis-CI, since it doesn't build this anyway.  Looks good!

**Pull request merged**